File: rb/BeginnersSelection/06_CardGameForTwo.rb
# https://atcoder.jp/contests/abs/tasks/abc088_b

class Object
  def let(f)
    f.call self
  end
end

[gets, gets][1]
  .split(" ")
  .map(&:to_i)
  .sort { |a, b| b <=> a }
  .reduce([0, 0, 0]) { |(a, b, i), x|
    i % 2 == 0 ?
      [a + x, b, i + 1] :
      [a, b + x, i + 1]
  }
  .let(->((a, b)) { a - b })
  .let(method(:puts))

# https://atcoder.jp/contests/abs/submissions/21520118

File: js/BeginnersSelection/06_CardGameForTwo.js
// https://atcoder.jp/contests/abs/tasks/abc088_b

Object.prototype.let = function(f) {
  return f(this.valueOf());
}

require('fs')
  .readFileSync('/dev/stdin', 'utf8')
  .split('\n')[1]
  .split(' ')
  .map(x => +x)
  .sort((a, b) => b - a)
  .reduce(([a, b], x, i) =>
    i % 2 == 0
      ? [a + x, b]
      : [a, b + x]
    , [0, 0]
  )
  .let(([a, b]) => a - b)
  .let(console.log);

// https://atcoder.jp/contests/abs/submissions/21475903
